code/modules/detectivework/evidence.dm

```diff
@@ -56,9 +56,12 @@
 	if(!isturf(I.loc)) //If it isn't on the floor. Do some checks to see if it's in our hands or a box. Otherwise give up.
 		if(I.loc.atom_storage) //in a container.
 			I.loc.atom_storage.remove_single(user, I, src)
-		if(!user.dropItemToGround(I))
+		if(!user.is_holding(I) || HAS_TRAIT(I, TRAIT_NODROP))
 			return
 
+	if(QDELETED(I))
+		return
```

Comment:
this should be at the very top of the proc, no?

Reply:
Not really. The point is precisely to make sure that after all the checks/stuff we've done to it it's not qdeleted. If I hadn't also changed the dropItemToGround(I) check putting it at the top wouldn't solve the issue.

Reply:
hm okay